Detections/OracleDatabaseAudit/OracleDBAuditDropManyTables.yaml

```diff
@@ -0,0 +1,31 @@
+id: b3aa0e5a-75a2-4613-80ec-93a1be3aeb8f
+name: OracleDBAudit - Multiple tables dropped in short time
+description: |
+  'Detects when user drops many tables in short period of time.'
+severity: Medium
+requiredDataConnectors:
+  - connectorId: OracleDatabaseAudit
+    dataTypes:
+      - Syslog
+queryFrequency: 30m
+queryPeriod: 50m
+triggerOperator: gt
+triggerThreshold: 0
+tactics:
+  - Impact
+query: |
+  let tbl_threshold = 10;
+  OracleDatabaseAuditEvent
+  | where isnotempty(DstUserName)
+  | where DbAction =~ 'DROP'
+  | extend TableName = replace(@'[,\(\)]', '', extract(@'(?i)SELECT(.*?)FROM\s(.*?)\s', 2, Action))
+  | where isnotempty(TableName)
+  | where tolower(TableName) != 'SELECT'

```

Comment:
Instead of using tolower() here, use `!~` which matches without case sensitivity. This is better for performance of the query.

Reply:
updated